packages/core/src/state-management/state-validation.ts

```diff
@@ -35,7 +36,7 @@ export class RealStateValidation implements StateValidation {
     } catch {
       throw new Error('Commit missing when loading schema document');
     }
-    return this.loadDocument(commitId).then((doc) => doc.content);
+    return this.loadDocument<TileDoctype<T>>(commitId).then((doc) => doc.content);
```

Comment:
Wow, schema validation should really take place within the TileDocumentHandler!

Reply:
Yep agreed.  I just noticed this and actually added this comment in my last commit: https://github.com/ceramicnetwork/js-ceramic/blob/afcf35426582bbc6aa0a5b2181feb5bf5c5016f9/packages/core/src/conflict-resolution.ts#L266

Reply:
FYI I just filed https://github.com/ceramicnetwork/js-ceramic/issues/1219 for this.  Could be a good starting issue for our upcoming new hire to get familiar with the codebase?